Change record page layout

ConnorSheremeta opened the pull request:
- Changed record page layout to reflect mock-up
- Added font-awesome gem
- Removed remaining glyphicons assets (removed from bootstrap in version 4)
#1426

ConnorSheremeta commented:
Note: this is based off of my upgrade_bootstrap branch due to it being a dependency of this change. Until that is merged, this PR will include all changes made from both branches (which will make it difficult to review.)

ConnorSheremeta commented:
Changes to record page layout are now live at http://uatsrv01.library.ualberta.ca:3012

I wanted to confirm in particular that the holdings table had proper layout and labels as pictured here @seanluyk :

![1678SS](https://user-images.githubusercontent.com/14242562/59876299-788a8c00-9360-11e9-98cc-99d94dc571da.png)

theLinkResolver commented:
@seanluyk No item type in the new display? Is that on purpose? I recommend caution as our item types include circulation policies such as Local Loan (this is terrible practice but it is what it is, without institutional and consortial investment in changing it).

seanluyk commented:
@theLinkResolver good question re: item type. If you take a look at NEOS discovery, item type is not included in the holdings table. I imagine that this was a conscious decision to simplify it, but we can always look at this again. The record view change is modelled on NEOS discovery, so that's why you aren't seeing it here. That being said, @ConnorSheremeta would you be able to add in item type below status for symphony records so we could see what it looks like?

seanluyk commented:
@ConnorSheremeta I'm really happy with the new and improved record layout, thanks so much for your work on this! In addition to adding in item type, I have a few other minor comments:
- In the tools menu, if it isn't too difficult, could we change the order of links to match NEOS Discovery (just for consistency): 
![image](https://user-images.githubusercontent.com/14880206/59952238-ae9b3f00-9438-11e9-820b-df164e6fa7a6.png)
- For the Kule holdings table, could the location be placed at the top of the box like in the SFX and Symphony tables, and the access label be removed?
 - Just an FYI that in our usage rights in SFX records, this is soon to be replaced (see #1315)

Other than that I think this is good to go, and I can document any other improvements in separate issues

ConnorSheremeta commented:
Addressed your comments @seanluyk , attached new screenshots below:
![Screenshot from 2019-06-24 15-50-52](https://user-images.githubusercontent.com/14242562/60055992-a5121100-969c-11e9-84a8-f124bbd3100c.png)
___
![Screenshot from 2019-06-24 15-08-55](https://user-images.githubusercontent.com/14242562/60055993-a5121100-969c-11e9-87bd-a85b5b9f6c1a.png)
___
![Screenshot from 2019-06-24 15-01-46](https://user-images.githubusercontent.com/14242562/60055995-a5121100-969c-11e9-8e67-b29bd9f005ea.png)

seanluyk commented:
Looking good, thanks for making these changes @ConnorSheremeta! After talking with @weiweishi today, we think that it would be a good idea for @nnunn to review this PR so we have another set of (UX focused) eyes on it

weiweishi commented:
@ConnorSheremeta @seanluyk I have tagged @nnunn as a reviewer for the PR. The code looks good to me.

ConnorSheremeta commented:
Added fix for #1488, once this is merged I'll close that issue

pgwillia commented on app/views/catalog/_holdings.html.erb:
```diff
@@ -1,72 +1,70 @@
+                    <p style="font-size: small">
+                      <strong>
+                        Note: For access, please contact any University of Alberta Libraries service desk in person or at
+                        <a href="http://www.library.ualberta.ca/askus" target="_blank">
```
```suggestion
                        <%= link_to 'Ask Us', cms_root_url('ask-us'), target: '_blank' %>
```

pgwillia commented on app/views/catalog/_holdings.html.erb:
```diff
@@ -1,72 +1,70 @@
+                      <strong>
+                        Note: For access, please contact any University of Alberta Libraries service desk in person or at
+                        <a href="http://www.library.ualberta.ca/askus" target="_blank">
+                          www.library.ualberta.ca/askus.
```
```suggestion
                         <%= cms_root_url('ask-us') %>
```

pgwillia commented on app/views/catalog/_holdings.html.erb:
```diff
@@ -91,34 +89,50 @@
+            <!--Subscription-->
+            <span class="where-is-this-item">
+              <strong><%= item[1][:name] %></strong>
+              <a href="<%= item[1][:url] %>" target='_blank'><%= icon('fas', 'info-circle') %></a>
```
Opportunity to use link_to here.

pgwillia commented on app/views/catalog/_holdings.html.erb:
```diff
@@ -130,30 +144,34 @@
+            <!--Access-->
+            <span class="where-is-this-item">
+              <strong><%= item[:access_title] %></strong>
+              <a href="<%= item[:access_url] %>" target="_blank"><%= icon('fas', 'info-circle') %></a>
```
Another opportunity to use link_to

pgwillia commented on app/views/catalog/_show_sidebar.html.erb:
```diff
@@ -6,55 +6,59 @@
-        <%= link_to 'Librarian View', librarian_view_catalog_path(id: @document), id: 'citeLink', class: 'nav-link lightboxLink' %>
-      </li>
+  <ul class="nav nav-stacked">
+    <li class= "nav-item"><a href="/bookmarks">Go to Bookmarks</a></li>
```
Another opportunity to use link_to

pgwillia commented on app/views/catalog/_show_sidebar.html.erb:
```diff
@@ -6,55 +6,59 @@
+      <%= link_to 'Send Correction', new_catalog_correction_path(catalog_id: @document['id']), class: 'nav-link' %>
+    </li>
+
+    <li class= "nav-item"><a href="/search_history">View Search History</a></li>
```
Another opportunity to use link_to

pgwillia commented on spec/features/catalog_search_spec.rb:
```diff
@@ -21,8 +21,8 @@
       expect(page).to have_text('Accountancy') # title
-      expect(page).to have_link('Business Source Complete') # Subscription
-      expect(page).to have_link('Factiva') # Subscription
+      expect(page).to have_text('Business Source Complete') # Subscription
```
This change brought something to my attention
![Screenshot from 2019-07-05 11-42-15](https://user-images.githubusercontent.com/1220762/60737997-f7afc080-9f19-11e9-9932-22c9cf497727.png)
![Screenshot from 2019-07-05 11-39-01](https://user-images.githubusercontent.com/1220762/60737998-f7afc080-9f19-11e9-9cf1-276edb76d411.png)
The link to 'Factiva' is now more hidden than before.  (Hint: it's under the information icon).  Is there another way to present this?

ConnorSheremeta commented:
I was modelling the mobile page after NEOS discovery (which has the left-sidebar first on mobile), it does make more sense though to have it after the main content which is now how it is displayed (screenshot below.) I've changed the links to use the link_to helper and I've changed the SFX subscription link back to being the text itself rather than under an information icon, which I think makes more sense (screenshot below.)

![Screenshot from 2019-07-08 12-22-34](https://user-images.githubusercontent.com/14242562/60833305-31333680-a17b-11e9-9e5a-820aaf0574ca.png)

seanluyk commented:
@ConnorSheremeta this is a great improvement, thanks!

weiweishi commented:
This looks good to me based on the comments from @pgwillia and changes @ConnorSheremeta has made.
